### PR TITLE
Each storage manager memorizes its own device id

### PR DESCRIPTION
**Pull request by Sotskin**
Also removed some old getmeminfo function call

### CODE REVIEW

File: src/storage/pooled_storage_manager.h [hunk 1]
```diff
@@ -99,34 +100,34 @@ class GPUPooledStorageManager final : public StorageManager {
...
   std::lock_guard<std::mutex> lock(Storage::Get()->GetMutex(Context::kGPU));
-  int device = handle->ctx.real_dev_id();
   size_t size = handle->size + NDEV;
   auto&& reuse_it = memory_pool_.find(size);
   if (reuse_it == memory_pool_.end() || reuse_it->second.size() == 0) {
     size_t free, total;
-    memory_manager_->MemGetInfo(device, &free, &total);
-    if (free <= total * reserve_ / 100 || size > free - total * reserve_ / 100) {
+    if (!memory_manager_->TryAllocate(device_id_, size + total * reserve_ / 100) || !memory_manager_->TryAllocate(device_id_, total * reserve_ / 100)) {
       do_reuse_= false;
```
Comment (fegin) [inline src/storage/pooled_storage_manager.h]: "do_reuse_ = false" . Missing one space.

File: src/storage/pooled_storage_manager.h [hunk 2]
```diff
@@ -99,34 +100,34 @@ class GPUPooledStorageManager final : public StorageManager {
   std::unordered_map<size_t, std::vector<void*>> memory_pool_;
   // whether to reuse freed memory
   bool do_reuse_;
+  // device id
```
Comment (fegin) [inline src/storage/pooled_storage_manager.h]: This command is redundant. The variable's naming provides enough information.

**Review by fegin — CHANGES_REQUESTED** [review]
Some nits. You can merge it after fixing them.

File: src/storage/pooled_storage_manager.h [hunk 3]
```diff
@@ -99,34 +100,34 @@ class GPUPooledStorageManager final : public StorageManager {
...
 void GPUPooledStorageManager::Alloc(Storage::Handle* handle) {
   std::lock_guard<std::mutex> lock(Storage::Get()->GetMutex(Context::kGPU));
-  int device = handle->ctx.real_dev_id();
   size_t size = handle->size + NDEV;
   auto&& reuse_it = memory_pool_.find(size);
   if (reuse_it == memory_pool_.end() || reuse_it->second.size() == 0) {
     size_t free, total;
-    memory_manager_->MemGetInfo(device, &free, &total);
-    if (free <= total * reserve_ / 100 || size > free - total * reserve_ / 100) {
+    if (!memory_manager_->TryAllocate(device_id_, size + total * reserve_ / 100) || !memory_manager_->TryAllocate(device_id_, total * reserve_ / 100)) {
```
Comment (fegin) [inline src/storage/pooled_storage_manager.h]: This line is too long. You can break it right after "||".

**Review by fegin — CHANGES_REQUESTED** [review]
Another nit.

**Review by qingsenzhang — APPROVED** [review]
Looks good

**Review by KarlLeell — APPROVED** [review]

**Pull request merged**